test/common/network/configurator_test.cc

```diff
@@ -0,0 +1,41 @@
+#include "test/extensions/common/dynamic_forward_proxy/mocks.h"
+
+#include "gtest/gtest.h"
+#include "library/common/network/configurator.h"
+
+using testing::_;
+using testing::Return;
+
+namespace Envoy {
+namespace Network {
+
+class ConfiguratorTest : public testing::Test {
```

Comment:
Do we also want an integration test?

Reply:
I have one, will push a separate branch in a bit.